cluster-autoscaler/cloudprovider/gce/gce_reserved_test.go

```diff
@@ -91,15 +92,31 @@ func TestCalculateKernelReservedLinux(t *testing.T) {
 			reservedMemory: 2*GiB + kernelReservedMemory + swiotlbReservedMemory,
 			osDistribution: OperatingSystemDistributionUbuntu,
 		},
+		{
+			physicalMemory: 3 * GiB,
+			reservedMemory: 48*MiB + kernelReservedMemory + lowMemoryOffset,
+			osDistribution: OperatingSystemDistributionCOS,
+			arch:           Arm64,
+		},
+		{
+			physicalMemory: 3.25 * GiB,
+			reservedMemory: 52*MiB + kernelReservedMemory + swiotlbReservedMemory + lowMemoryOffset,
+			osDistribution: OperatingSystemDistributionCOS,
+			arch:           Arm64,
+		},
 	}
 	for idx, tc := range testCases {
 		r := &GceReserved{}
 		t.Run(fmt.Sprintf("%v", idx), func(t *testing.T) {
-			reserved := r.CalculateKernelReserved(tc.physicalMemory, OperatingSystemLinux, tc.osDistribution, "", "")
+			reserved := r.CalculateKernelReserved(tc.physicalMemory, OperatingSystemLinux, tc.osDistribution, tc.arch, "")
 			if tc.osDistribution == OperatingSystemDistributionUbuntu {
 				assert.Equal(t, tc.reservedMemory+int64(math.Min(correctionConstant*float64(tc.physicalMemory), maximumCorrectionValue)+ubuntuSpecificOffset), reserved)
 			} else if tc.osDistribution == OperatingSystemDistributionCOS {
-				assert.Equal(t, tc.reservedMemory+int64(math.Min(correctionConstant*float64(tc.physicalMemory), maximumCorrectionValue)), reserved)
+				val := tc.reservedMemory + int64(math.Min(correctionConstant*float64(tc.physicalMemory), maximumCorrectionValue))
```

Comment:
I'm not sure what this is testing exactly. It looks like we're just repeating the actual calculation in the test code? If there's a bug in the calculation itself, how would we catch it? And what do these tests even give us? Normally I'd expect that `tc.reservedMemory` would be the expected end value that `reserved` would be compared against (possibly with some tolerations).

Reply:
I agree. The test scenario for reserved tests don't provide a lot of signal.
However, there isn't much "logic" per-say in reserved.go, and it is essentially hardcoded values with some conditions. So extracting signals from this isn't particularly useful. We can only test if the values / conditions in reserved.go don't change or flag unintended changes.

IIUC is the idea of having a "static" value for `reservedMemory`? essentially we would just be compressing the manual conditions and live calculations (`52*MiB + kernelReservedMemory + swiotlbReservedMemory + lowMemoryOffset`) to pre-calculated in the value. Not sure if there is value in that change?

Reply:
Yeah I guess it doesn't change much, and the current solution is in line with the existing tests.